src/types.rs

```diff
@@ -83,6 +83,12 @@ pub trait AsnType {
     fn is_present(&self) -> bool {
         true
     }
+    /// Whether the type is choice type.
+    ///
+    /// Custom implementation is only used for `CHOICE` type.
+    fn is_choice() -> bool {
```

Comment:
Should this be an associated constant rather than a function? It seems like something that doesn't needs to be a function.